Default cache path for rstan model causes rbmi to crash when run by the same user in parallel / inconsistent implement and comment

**Describe the bug**
The default cache path for rstan model causes rbmi to crash when run by the same user in parallel. This is caused by rstan recompiling the model each time (and race conditions occurring when this is done in parallel).

**To Reproduce**
Run the rbmi vignette on several cores using the same user account in parallel. 
I saw that in rbmi 1.3.1 the model file is now using a hash, which should also help with this (see #459)

**Environment (please complete the following information):**
 - OS: several
 - R version 4.4.1
 - rbmi version 1.3.0

**Proposed solution**
My proposal to fix this permanently is to change the default in https://github.com/insightsengineering/rbmi/blob/1d150e601a8191de4fa632599f4ededc7ad176d1/R/utilities.R#L620 to reflect the comment  

> #' Directory to store compiled Stan model in. If not set, a temporary directory is used for
> #' the given R session. Can also be set via the environment variable `RBMI_CACHE_DIR`.

That is, to set the default to `tempdir(check = TRUE)`, which is the _per-session temporary directory_ instead of the current default `tools::R_user_dir("rbmi", which = "cache")`, which is a _persistent_ per-user cache directory.

The workaround for the current version is to set `options("rbmi.cache_dir" = tempdir(check = TRUE))`.

Tagging @bailliem for awareness

@gowerc, on my Mac, changing your example to use 12 cores causes this error almost immediately:

```
1 error generated.
make: *** [foo.o] Error 1
1 error generated.
make: *** [foo.o] Error 1
Stan model 'rbmi_mmrm' does not contain samples.
Stan model 'rbmi_mmrm' does not contain samples.
Stan model 'rbmi_mmrm' does not contain samples.
Stan model 'rbmi_mmrm' does not contain samples.
Stan model 'rbmi_mmrm' does not contain samples.
Stan model 'rbmi_mmrm' does not contain samples.
[1] "hi"
[1] "hi"
Error in checkForRemoteErrors(val) : 
  24 nodes produced errors; first error: 0
Syntax error in 'string', line 73, column 0 to column 9, parsing error:
   -------------------------------------------------
    71:     vector[P] beta = R_inverse * theta;
    72:  }
    73:  functions {
         ^
    74:      int integer_division(int a, int b) {
    75:          // perform a/b ensuring return value is also an int
   -------------------------------------------------

Expected end of file after end of generated quantities block.
```

--- a/R/utilities.R
+++ b/R/utilities.R
@@ -574,12 +574,12 @@ get_stan_model <- function() {
     cache_dir <- getOption("rbmi.cache_dir")
     dir.create(cache_dir, showWarnings = FALSE, recursive = TRUE)
     model_file <- file.path(cache_dir, paste0("MMRM_", get_session_hash(), ".stan"))
-    
+
     if (!file.exists(model_file)) {
         clear_model_cache()
         file.copy(file_loc, model_file, overwrite = TRUE)
     }
-    
+
     rstan::stan_model(
         file = model_file,
         auto_write = TRUE,
@@ -603,8 +603,18 @@ get_stan_model <- function() {
 #'
 #' Default = `tools::R_user_dir("rbmi", which = "cache")`
 #'
-#' Directory to store compiled Stan model in. If not set, a temporary directory is used for
-#' the given R session. Can also be set via the environment variable `RBMI_CACHE_DIR`.
+#' Directory to store compiled Stan models in to avoid having to re-compile.
+#' If the environment variable `RBMI_CACHE_DIR` has been set this will be used
+#' as the default value.
+#' Note that if you are running rbmi in multiple R processes at the same time
+#' (that is say multiple calls to `Rscript` at once) then there is a theoretical
+#' risk of the processes breaking each other as they attempt to read/write to the
+#' same cache folder at the same time. To avoid this potential issue it is recommended
+#' to set the cache directory to a unique folder for each R session e.g.
+#'
+#' ```
+#' options("rbmi.cache_dir" = tempdir(check = TRUE))
+#' ```
 #'
 #'
 #' @examples

--- a/man/rbmi-settings.Rd
+++ b/man/rbmi-settings.Rd
@@ -18,8 +18,17 @@ Each of the following are the name of options that can be set via:
 
 Default = \code{tools::R_user_dir("rbmi", which = "cache")}
 
-Directory to store compiled Stan model in. If not set, a temporary directory is used for
-the given R session. Can also be set via the environment variable \code{RBMI_CACHE_DIR}.
+Directory to store compiled Stan models in to avoid having to re-compile.
+If the environment variable \code{RBMI_CACHE_DIR} has been set this will be used
+as the default value.
+Note that if you are running rbmi in multiple R processes at the same time
+(that is say multiple calls to \code{Rscript} at once) then there is a theoretical
+risk of the processes breaking each other as they attempt to read/write to the
+same cache folder at the same time. To avoid this potential issue it is recommended
+to set the cache directory to a unique folder for each R session e.g.
+
+\if{html}{\out{<div class="sourceCode">}}\preformatted{options("rbmi.cache_dir" = tempdir(check = TRUE))
+}\if{html}{\out{</div>}}
 }
 }
 \examples{

